container-chains/nodes/simple/src/command.rs

```diff
@@ -297,20 +300,89 @@ pub fn run() -> Result<()> {
+                        .spawn_handle()
+                        .spawn("rpc_provider_exemple", None, async move {
+                            let mut stream = client.new_best_notification_stream().await.unwrap();
+
+                            while let Some(header) = stream.next().await {
+                                log::info!("New best block: {}", header.hash());
+                            }
+                        });
+                }
+
+                // POC: Spawn container chain embed node
+                {
+                    let collator_options = cli.run.collator_options();
+
+                    let polkadot_cli = RelayChainCli::new(
+                        &config,
+                        [RelayChainCli::executable_name()]
+                            .iter()
+                            .chain(cli.relay_chain_args.iter()),
+                    );
+
+                    let tokio_handle = config.tokio_handle.clone();
+                    let polkadot_config = SubstrateCli::create_configuration(
+                        &polkadot_cli,
+                        &polkadot_cli,
+                        tokio_handle,
+                    )
+                    .map_err(|err| format!("Relay chain argument error: {}", err))?;
+
+                    let telemetry = config
+                        .telemetry_endpoints
+                        .clone()
+                        .filter(|x| !x.is_empty())
+                        .map(|endpoints| -> std::result::Result<_, sc_telemetry::Error> {
+                            let worker = TelemetryWorker::new(16)?;
+                            let telemetry = worker.handle().new_telemetry(endpoints);
+                            Ok((worker, telemetry))
+                        })
+                        .transpose()
+                        .map_err(sc_service::Error::Telemetry)?;
+
+                    let telemetry_worker_handle =
+                        telemetry.as_ref().map(|(worker, _)| worker.handle());
+
+                    let (relay_chain_interface, _collation_pair) = build_relay_chain_interface(
+                        polkadot_config,
+                        &config,
+                        telemetry_worker_handle,
+                        &mut task_manager,
+                        collator_options,
+                        None,
+                    )
+                    .await
+                    .map_err(|e| sc_service::Error::Application(Box::new(e) as Box<_>))?;
+
+                    let keystore_container = sc_service::KeystoreContainer::new(
+                        &sc_service::config::KeystoreConfig::InMemory,
+                    )?;
+                    let keystore = keystore_container.keystore();
+
+                    let para_id = cli.run.parachain_id.unwrap_or(2000);
+
+                    let (_cc_task_manager, _cc_client, _cc_backend) =
+                        tc_service_container_chain::service::start_node_impl_container(
```

Comment:
Question about this, we probably need to start this with very different arguments for frontier-like nodes anhd non-frontier-like nodes (specially related to the rpc-nodes). Are we going to be able to do so?

Reply:
Or maybe we can wire the ethereum ones to their own rpc-setup, but we for sure need to make sure not to spawn a non-ethereum registered chain in that case.

So `start_node_impl_container` is likely usable for substrate chains + collators but not for ethereum-nodes